protob/messages-eos.proto

```diff
@@ -0,0 +1,265 @@
+    optional bool show_display = 2; // optionally show on display before sending the result
+}
+ /**
+ * Response: Contains an Eos public key derived from device private seed
+ * @end
+ */
+ message EosPublicKey {
+    optional string wif_public_key = 1;     // EOS pub key in Base58 encoding
+    optional bytes raw_public_key = 2;      // Raw public key
+}
+ /**
+ * Request: Ask device to sign transaction
+ * @start
+ * @next EosTxRequest
+ * @next Failure
+ */
+ message EosSignTx {
+    repeated uint32 address_n = 1;                // BIP-32 path to derive the key from master node 44'/194'/0'
+    optional bytes chain_id = 2;                  // 256-bit long chain id
+    optional EosTxHeader header = 3;              // EOS transaction header
+    optional uint32 num_actions = 4;              // number of actions 
+     /**
+    * Structure representing EOS transaction header
+    */
+    message EosTxHeader {
+        required uint32 expiration = 1;           // time at which transaction expires
+        required uint32 ref_block_num = 2;        // 16-bit specifies a block num in the last 2^16 blocks.
+        required uint32 ref_block_prefix = 3;     // specifies the lower 32 bits of the blockid at get_ref_blocknum
+        required uint32 max_net_usage_words = 4;  // upper limit on total network bandwidth (in 8 byte words) billed for this transaction
+        required uint32 max_cpu_usage_ms = 5;     // 8-bit upper limit on the total CPU time billed for this transaction
+        required uint32 delay_sec = 6;            // number of seconds to delay this transaction for during which it may be canceled.
+    }   
+}
+ /**
+ * Response: Device asks to upload next action
+ * @next EosTxActionAck
+ */
+message EosTxActionRequest {
+    optional uint32 data_size = 1;
+}
+ /**
+ * Request: Next action data that needs to be uploaded
+ * @next EosTxActionRequest
+ * @next EosSignedTx
+ * @next Failure
+ */
+message EosTxActionAck {
+    optional EosActionCommon common = 1;
+    optional EosActionTransfer transfer = 2;
+    optional EosActionDelegate delegate = 3;
+    optional EosActionUndelegate undelegate = 4;
+    optional EosActionRefund refund = 5;
+    optional EosActionBuyRam buy_ram = 6;
+    optional EosActionBuyRamBytes buy_ram_bytes = 7;
+    optional EosActionSellRam sell_ram = 8;
+    optional EosActionVoteProducer vote_producer = 9;     
+    optional EosActionUpdateAuth update_auth = 10;   
+    optional EosActionDeleteAuth delete_auth = 11;
+    optional EosActionLinkAuth link_auth = 12;
+    optional EosActionUnlinkAuth unlink_auth = 13;
+    optional EosActionNewAccount new_account = 14;
+    optional EosActionUnknown unknown = 15;
+     /**
+    * Structure representing asset type
+    */
+    message EosAsset {
+        optional sint64 amount = 1;
+        optional uint64 symbol = 2;           // First 8-bits used for precission.    
```

Comment:
shouldn't this read "_lowest_" 8 bits?

Reply:
also precision not precission

Reply:
I have fixed comments, And also did a rebase and squash a few minutes ago. Also for other pull requests i have also performed rebase on master and squash for commits.

Reply:
@matejcik Do you accept @tarassh points\explanation regarding Unknown action, that that's actually equivalent for TX payload in Ethereum integration? Probably the name is a bit misleading. as it's actually not unknown action, as it should be known by EOS ecosystem, we just cannot identify this action as a particular one on the device. If the name is confusing, we can rename it to something more meaningful, like EosActionPayload. So what do you think?